spec_parser/parser.py

```diff
@@ -268,6 +274,28 @@ def maybe_ext_properties(self, p):
     def ext_properties(self, p):
         return p.properties_list
 
+    @_("empty", "format")
+    def maybe_format(self, p):
+        if p[0] is None:
+            return []
+        return p[0]
+
+    @_("FORMAT format_pattern")
+    def format(self, p):
+        return p.format_pattern
+
+    @_("ULISTA")
+    def format_pattern(self, p):
+        ulista = p.ULISTA
+
+        # strip the md list identifier, ie r'[-*+]'
+        ulista = re.split(r"[-*+]", ulista, 1)[-1].strip()
+
+        # strip the pattern key to get only the specified regex
+        format_regex = re.split(r"pattern:\s", ulista, 1)[1]
```

Comment:
Please don't add such logic in the parsing code.

Exactly like metadata section, the parsing should return a list of attribute value pairs (specified in something like `valid_metadata_keys`) and then the caller should save the data they are interested in (in this case `pattern`)

Reply:
Okay, I thought the "Format" section does only have this one value for "pattern". But of course if we implement this more generic we are more flexible to changes in the future. I think this results in exactly the same code as for metadata except the check for valid values. So I would reuse the methods `metadata_list` and `metadata_line` and rename them to something more generic like `list_of_attribute_value_pairs`. What would be the best way to check if the parsed section is `metadata` or `format`? My attempt would be to use `p[-3]` but this also seems a bit hacky, do you have a better idea?